plugin/validator_test.go

```diff
@@ -175,6 +179,35 @@ func TestValidator_ProcessAutoFields(t *testing.T) {
 	})
 }
 
+func TestValidator_SchemaAsJSON(t *testing.T) {
+	schemaNames := []string{
+		"key-auth",
+		"rate-limiting",
+	}
+	v, err := NewValidator()
+	assert.Nil(t, err)
+	for _, schemaName := range schemaNames {
+		schema, err := ioutil.ReadFile("testdata/" + schemaName + ".lua")
+		assert.Nil(t, err)
+		assert.Nil(t, v.LoadSchema(string(schema)))
+	}
+
+	t.Run("returns a valid JSON schema for loaded plugin schema", func(t *testing.T) {
+		for _, schemaName := range schemaNames {
+			schema, err := v.SchemaAsJSON(schemaName)
+			assert.Nil(t, err)
+			var kongPluginConfig KongPluginConfig
+			assert.Nil(t, json.Unmarshal([]byte(schema), &kongPluginConfig))
+		}
+	})
+
+	t.Run("returns error when specifying unknown plugin", func(t *testing.T) {
+		schema, err := v.SchemaAsJSON("invalid-plugin")
+		assert.Error(t, err, "no plugin named 'invalid-plugin'")
+		assert.Empty(t, schema)
+	})
+}
```

Comment:
Add tests to assert the entire schema of the plugin.
Also, add a plugin with a function() and foreign references to ensure that the code introduced in the patch is exercised.